src/views/search/search-view.tsx

```diff
@@ -143,27 +115,48 @@ const SearchView: FC<SearchProps> = ({ useDisableSearch, useQuery, ResultsHeader
 			});
 
 			if (count > 0) {
-				setLoading(true);
+				// setLoading(true);
 				updateQuery(modifiedQuery);
 			}
 		}
-		if (query.length === 0) {
-			// setSearchResults(emptySearchResults);
+	}, [
+		searchResults,
+		queryArray,
+		updateQuery,
+		findIcon,
+		isInvalidQuery,
+		emptySearchResults,
+		query,
+		queryToString
+	]);
+
+	const loading = useMemo(() => {
+		if (searchResults.status === 'pending') {
+			return true;
 		}
-	}, [query, searchResults, queryArray, updateQuery, findIcon, isInvalidQuery, emptySearchResults]);
+		return false;
+	}, [searchResults.status]);
 
 	useEffect(() => {
-		if (query && query.length > 0 && query !== searchResults.query && !isInvalidQuery) {
-			setLoading(true);
+		if (query && query.length > 0 && queryToString !== searchResults.query && !isInvalidQuery) {
 			setFilterCount(query.length);
-			search(query, true);
+			searchQuery(queryToString, true);
 		}
 		if (query && query.length === 0) {
 			setFilterCount(0);
 			setIsInvalidQuery(false);
-			setResultLabel(t('label.results_for', 'Results for: '));
 		}
-	}, [query, search, searchResults.query, queryArray, t, isInvalidQuery]);
+	}, [query, queryArray, t, isInvalidQuery, searchQuery, searchResults.query, queryToString]);
+
+	const filteredSearchResults = useMemo(() => {
+		const conversations = filter(searchResults.conversations, (conv) => {
+			if (conv.parent === FOLDERS.TRASH) {
+				return false;
+			}
+			return true;
+		});
+		return { ...searchResults, conversations };
+	}, [searchResults]);
```

Comment:
I guess there is setting for it,also since it's a search .So user may want to find the deleted msg/conv